### PR TITLE
add tests for single visualization base

### PR DESCRIPTION
**Pull request by hainv-ohio**

### CODE REVIEW

File: src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            // Assert
+            Assert.Equal(title, visualization.Title);
+            Assert.Equal(dataSourceItem, visualization.DataDefinition.DataSourceItem);
+            Assert.IsType<SingleValueLabelsVisualizationDataSpec>(visualization.VisualizationDataSpec);
+        }
+
+        [Theory]
+        [InlineData("SingleValueLabelsVisualizationDataSpec")]
+        [InlineData("CategoryVisualizationDataSpec")]
+        public void VisualizationDataSpec_GetSetAsExpected_WhenUse(String type)
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs:34]: Update UT name

Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs:34]: Not sure if it's needed as this field only has default getter and setter.

File: src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            };
+            visualization.VisualizationDataSpec = visSpec;
+
+            // Act
+            var labels = visualization.Labels;
+
+            // Assert
+            Assert.NotNull(labels);
+            Assert.Equal(2, labels.Count);
+            Assert.Equal(visSpec.Rows, labels);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs:79]: I think we only need the final Assertions `Assert.Equal(visSpec.Rows, labels);`, as this includes other assertions

File: src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            var visualization = new TestSingleValueLabelsVisualization("testTitle", dataSourceItem);
+            var visSpec = new CategoryVisualizationDataSpec();
+            visualization.VisualizationDataSpec = visSpec;
+
+            // Act
+            var labels = visualization.Labels;
+
+            // Assert
+            Assert.NotNull(labels);
+            Assert.Empty(labels);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs:96]: Assert.Empty already includes Assert.NotNull

File: src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            };
+            visualization.VisualizationDataSpec = visSpec;
+
+            // Act
+            var values = visualization.Values;
+
+            // Assert
+            Assert.NotNull(values);
+            Assert.Equal(2, values.Count);
+            Assert.Equal(visSpec.Value, values);
```
Comment (vulh-infragistics) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs:121]: Assert.Equal(visSpec.Value, values); already covers 2 assertion above

**Review by vulh-infragistics — CHANGES_REQUESTED** [review]
Please update the comments

File: src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+            var dataSourceItem = new DataSourceItem();
+            var visualization = new TestSingleValueLabelsVisualization("testTitle", dataSourceItem);
+            var visSpec = new CategoryVisualizationDataSpec();
+            visualization.VisualizationDataSpec = visSpec;
+
+            // Act
+            var labels = visualization.Labels;
+
+            // Assert
+            Assert.Empty(labels);
```
Comment (recca5p) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs:67]: I think we should add an assertion to verify that the returned list is not null, even if empty

File: src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+            var dataSourceItem = new DataSourceItem();
+            var visualization = new TestSingleValueLabelsVisualization("testTitle", dataSourceItem);
+            var visSpec = new CategoryVisualizationDataSpec();
+            visualization.VisualizationDataSpec = visSpec;
+
+            // Act
+            var values = visualization.Values;
+
+            // Assert
+            Assert.Empty(values);
```
Comment (recca5p) [inline src/Reveal.Sdk.Dom.Tests/Visualizations/SingleValueLabelsVisualizationBaseFixture.cs:106]: Similar to the Labels test, add Assert.NotNull(values) to make sure that returned list is not null

**Review by recca5p — CHANGES_REQUESTED** [review]

**Review by recca5p — APPROVED** [review]

**Review by vulh-infragistics — APPROVED** [review]

**Pull request merged**